File: examples/typescript/basic.ts
/**
 * Basic Nexi AI Usage - TypeScript
 *
 * This example shows basic usage of the Nexi AI API.
 * Run with: npx tsx examples/typescript/basic.ts
 */

import { NexiAPI } from '../../src/api/index.js';

async function main() {
  console.log('🤖 Nexi AI - Basic Example\n');

  // Create Nexi instance with custom config
  const nexi = await NexiAPI.create({
    model: 'llama3.1:8b',
    dataDir: './example-data',
    logLevel: 'info',
  });

  // Check if Ollama is available
  const available = await nexi.isAvailable();
  if (!available) {
    console.log('❌ Ollama is not running. Please start it with: ollama serve');
    return;
  }

  console.log('✅ Connected to Ollama\n');

  // Simple chat
  console.log('You: Hello!');
  const response1 = await nexi.chat('Hello!');
  console.log(`Nexi: ${response1.message}`);
  console.log(`  (mood: ${response1.mood}, energy: ${response1.energy})\n`);

  // Another message
  console.log('You: What do you like to do for fun?');
  const response2 = await nexi.chat('What do you like to do for fun?');
  console.log(`Nexi: ${response2.message}\n`);

  // Store a memory
  nexi.remember('User is learning to use Nexi AI', { importance: 8 });
  console.log('📝 Stored memory: "User is learning to use Nexi AI"\n');

  // Get stats
  const stats = nexi.getStats();
  console.log('📊 Stats:');
  console.log(`  Mode: ${stats.mode}`);
  console.log(`  Mood: ${stats.mood}`);
  console.log(`  Energy: ${stats.energy}`);
  console.log(`  Interactions: ${stats.interactions}`);
  console.log(`  Memories: ${stats.memories.total}`);

  // Cleanup
  nexi.shutdown();
  console.log('\n✨ Done!');
}

main().catch(console.error);

File: examples/python/basic.py
"""
Basic Nexi AI Usage - Python

This example shows basic usage of the Nexi AI Python client.
Run with: python examples/python/basic.py
"""

import sys
sys.path.insert(0, './python')

from nexi import Nexi, NexiConfig

def main():
    print("🤖 Nexi AI - Basic Example\n")

    # Create Nexi instance with custom config
    config = NexiConfig(
        model="llama3.1:8b",
        data_dir="./example-data-py",
    )
    nexi = Nexi(config)

    # Check if Ollama is available
    if not nexi.is_available():
        print("❌ Ollama is not running. Please start it with: ollama serve")
        return

    print("✅ Connected to Ollama\n")

    # Simple chat
    print("You: Hello!")
    response1 = nexi.chat("Hello!")
    print(f"Nexi: {response1.message}")
    print(f"  (mood: {response1.mood.value}, energy: {response1.energy.value})\n")

    # Another message
    print("You: What do you like to do for fun?")
    response2 = nexi.chat("What do you like to do for fun?")
    print(f"Nexi: {response2.message}\n")

    # Store a memory
    nexi.remember("User is learning to use Nexi AI", importance=8)
    print('📝 Stored memory: "User is learning to use Nexi AI"\n')

    # Get stats
    stats = nexi.get_stats()
    print("📊 Stats:")
    print(f"  Mode: {stats.mode.value}")
    print(f"  Mood: {stats.mood.value}")
    print(f"  Energy: {stats.energy.value}")
    print(f"  Interactions: {stats.interactions}")
    print(f"  Memories: {stats.total_memories}")

    # Cleanup
    nexi.shutdown()
    print("\n✨ Done!")


if __name__ == "__main__":
    main()
